### DISCUSSION TITLE
Remplace BT communication by CAN or RS485

### DISCUSSION THREAD

**Discussion by ttiburst** [fl4p/batmon-ha#212] [Q&A]
I have JK BMS anD BT is not stable, let me know if it is possible to replace the BT communication by a RS485 or better a CAN communication.

Thanks.

**Comment by typxxi** [comment]
have you found a solution ?
Same issue here ... it works and then it does not like here when HA was fully working - only bat mon data missing.

![image](https://github.com/user-attachments/assets/fa45d345-92fe-4942-955f-cc2f7e8c7bf8)

**Comment by typxxi** [comment]
for those who are searching 

here is a link to an esp solution connecting the battery via esp and wifi to HA.

And in this particular comment from the developer he also links to a project that runs esp 32 and connects the 2nd RS485 port used for internal communication between the BMS ses of serveral batteries in parallel.
Yes, it is possible to use CAN for inverter communication, then RS485 for communication with the windows JK BMS App and finally the RS485-2 port. 

https://github.com/syssi/esphome-jk-bms/discussions/715#discussioncomment-12649276
